array/flatten.py

```diff
@@ -1,26 +1,36 @@
+from collections import Iterable
+from types import GeneratorType
```

Comment:
`Iterable` is fine but I think the previous `collections.Iterable` was better here.  
The `GeneratorType` is not needed, `yield` clearly specifies a `generator`.